src/core/lib/surface/call.cc

```diff
@@ -1441,6 +1441,10 @@ static grpc_call_error call_start_batch(grpc_call* call, const grpc_op* ops,
           error = GRPC_CALL_ERROR_INVALID_METADATA;
           goto done_with_error;
         }
+        if (call->send_initial_metadata.get(grpc_core::TeMetadata()) ==
```

Comment:
could we simplify this to `call->send_initial_metadata.Remove(grpc_core::TeMetadata());` - ie do it unilaterally

Reply:
sure just for initial metadata or for both initial and trailing metadata ?

Reply:
![image](https://user-images.githubusercontent.com/10120821/152239536-8fec5fff-b70c-4949-8448-7708652dee21.png)

Reply:
lol 👍